diff --git a/configure b/configure
index 0edea88..80ddd19 100755
--- a/configure
+++ b/configure
@@ -27830,7 +27830,8 @@ fi
 
     # Test that the package found is for the right architecture
     saved_LIBS="$LIBS"
-    LIBS="$OPENJPEG_LIBS"
+    # Bug in the pkg-config file? Have to link pthread/libm explicitly for the check to pass
+    LIBS="$OPENJPEG_LIBS -lm -pthread"
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for opj_setup_decoder in -lopenjp2" >&5
 $as_echo_n "checking for opj_setup_decoder in -lopenjp2... " >&6; }
 if ${ac_cv_lib_openjp2_opj_setup_decoder+:} false; then :
@@ -31639,7 +31640,7 @@ fi
 
             unset ac_cv_lib_sqlite3_sqlite3_open
             saved_LIBS="$LIBS"
-            LIBS=""
+            LIBS="-pthread -ldl"
             { $as_echo "$as_me:${as_lineno-$LINENO}: checking for sqlite3_open in -lsqlite3" >&5
 $as_echo_n "checking for sqlite3_open in -lsqlite3... " >&6; }
 if ${ac_cv_lib_sqlite3_sqlite3_open+:} false; then :
@@ -32024,7 +32025,7 @@ fi
 
             unset ac_cv_lib_sqlite3_sqlite3_open
             saved_LIBS="$LIBS"
-            LIBS=""
+            LIBS="-pthread -ldl"
             { $as_echo "$as_me:${as_lineno-$LINENO}: checking for sqlite3_open in -lsqlite3" >&5
 $as_echo_n "checking for sqlite3_open in -lsqlite3... " >&6; }
 if ${ac_cv_lib_sqlite3_sqlite3_open+:} false; then :
@@ -33486,7 +33487,7 @@ $as_echo "$as_me: WARNING: GEOS was found on your system, but geos-config report
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c -lgeos"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
@@ -33495,6 +33496,12 @@ $as_echo "$as_me: WARNING: GEOS was found on your system, but geos-config report
       ax_save_CFLAGS="${CFLAGS}"
       CFLAGS="${GEOS_CFLAGS}"
 
+      ac_ext=cpp
+ac_cpp='$CXXCPP $CPPFLAGS'
+ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
+
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GEOSversion in -lgeos_c" >&5
 $as_echo_n "checking for GEOSversion in -lgeos_c... " >&6; }
 if ${ac_cv_lib_geos_c_GEOSversion+:} false; then :
@@ -33521,7 +33528,7 @@ return GEOSversion ();
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
+if ac_fn_cxx_try_link "$LINENO"; then :
   ac_cv_lib_geos_c_GEOSversion=yes
 else
   ac_cv_lib_geos_c_GEOSversion=no
@@ -33538,6 +33545,12 @@ else
   HAVE_GEOS="no"
 fi
 
+      ac_ext=c
+ac_cpp='$CPP $CPPFLAGS'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compiler_gnu=$ac_cv_c_compiler_gnu
+
 
       if test x"$HAVE_GEOS" = "xno"; then
           GEOS_CFLAGS=""
diff --git a/configure.ac b/configure.ac
index b6e035d..34ff93e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2757,7 +2757,8 @@ else
 
     # Test that the package found is for the right architecture
     saved_LIBS="$LIBS"
-    LIBS="$OPENJPEG_LIBS"
+    # Bug in the pkg-config file? Have to link pthread/libm explicitly for the check to pass
+    LIBS="$OPENJPEG_LIBS -lm -pthread"
     AC_CHECK_LIB(openjp2,opj_setup_decoder, [HAVE_OPENJPEG=yes], [HAVE_OPENJPEG=no])
     LIBS="$saved_LIBS"
 
diff --git a/m4/ax_lib_sqlite3.m4 b/m4/ax_lib_sqlite3.m4
index 477d4e2..52ea6d1 100644
--- a/m4/ax_lib_sqlite3.m4
+++ b/m4/ax_lib_sqlite3.m4
@@ -76,7 +76,7 @@ AC_DEFUN([AX_LIB_SQLITE3],
 
             unset ac_cv_lib_sqlite3_sqlite3_open
             saved_LIBS="$LIBS"
-            LIBS=""
+            LIBS="-pthread -ldl"
             AC_CHECK_LIB(sqlite3,sqlite3_open,LIB_SQLITE3_FOUND=yes,LIB_SQLITE3_FOUND=no,-L$ac_sqlite3_path/lib)
             LIBS="$saved_LIBS"
             if test "$LIB_SQLITE3_FOUND" = "yes"; then
diff --git a/m4/geos.m4 b/m4/geos.m4
index a410673..726bd5f 100644
--- a/m4/geos.m4
+++ b/m4/geos.m4
@@ -121,7 +121,7 @@ AC_DEFUN([GEOS_INIT],[
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c -lgeos"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
@@ -130,12 +130,14 @@ AC_DEFUN([GEOS_INIT],[
       ax_save_CFLAGS="${CFLAGS}"
       CFLAGS="${GEOS_CFLAGS}"
 
+      AC_LANG_PUSH([C++])
       AC_CHECK_LIB([geos_c],
         [GEOSversion],
         [HAVE_GEOS="yes"],
         [HAVE_GEOS="no"],
         []
       )
+      AC_LANG_POP([C++])
 
       if test x"$HAVE_GEOS" = "xno"; then
           GEOS_CFLAGS=""
